pytests/tests/test_checksum.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (C) 2019 - 2022 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+
+import os
+import shutil
+import re
+import filecmp
+import pytest
+
+WORKDIR = '/root/repofrompath/workdir'
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    yield
+    teardown_test(utils)
+
+
+def teardown_test(utils):
+    utils.run(['tdnf', 'erase', '-y', 'tdnf-test-one'])
+    if os.path.isdir(WORKDIR):
+        shutil.rmtree(WORKDIR)
+
+
+def enable_and_create_repo(utils):
+    workdir = WORKDIR
+    utils.makedirs(workdir)
+    reponame = 'photon-test-sha512'
+
+    ret = utils.run(['tdnf', '-v', '--disablerepo=*', '--enablerepo=photon-test-sha512', 'makecache'])
+    assert ret['retval'] == 0
+
+    ret = utils.run(['tdnf', '--repo={}'.format(reponame),
+                     '--download-metadata',
+                     'reposync'],
+                    cwd=workdir)
+    assert ret['retval'] == 0
+    synced_dir = os.path.join(workdir, reponame)
+    assert os.path.isdir(synced_dir)
+    assert os.path.isdir(os.path.join(synced_dir, 'repodata'))
+    assert os.path.isfile(os.path.join(synced_dir, 'repodata', 'repomd.xml'))
+
+
+def find_suffix(filename, prefix):
+    escaped_prefix = re.escape(prefix)
+
+    pattern = rf'^{escaped_prefix}(.+)$'
+
+    match = re.search(pattern, filename)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+
+def modify_rpm(workdir, orig_pkg, new_pkg):
+    orig_filename = ''
+    orig_path = ''
+
+    for filename in os.listdir(workdir):
+        if filename.startswith(orig_pkg):
+            suffix = find_suffix(filename, orig_pkg)
+
+            if suffix is not None:
+                orig_filename = orig_pkg + suffix
+                orig_path = os.path.join(workdir, filename)
+
+        elif filename.startswith(new_pkg):
+            suffix = find_suffix(filename, new_pkg)
+
+            if not os.path.exists(orig_path):
+                orig_filename = orig_pkg + '-1.0.1-2.x86_64.rpm'
+                orig_path = os.path.join(workdir, orig_filename)
+                file = open(orig_path, 'w')
+                file.close()
+
+            if suffix is not None:
+                new_path = os.path.join(workdir, filename)
+
+                if not filecmp.cmp(new_path, orig_path):
+                    shutil.copy2(new_path, orig_path)
+
+
+# install package with SHA512
+def test_install_package_with_sha512_checksum(utils):
+    workdir = WORKDIR
+    reponame = 'photon-test-sha512'
+    synced_dir = os.path.join(workdir, reponame)
+
+    enable_and_create_repo(utils)
```

Comment:
Since you are not modifying the repo for this test, you can just use `photon-test-sha512` directly.